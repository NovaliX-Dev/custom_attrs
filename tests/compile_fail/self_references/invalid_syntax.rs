use custom_attrs::CustomAttrs;

#[derive(CustomAttrs)]
#[attr(pub a: usize)]
enum Enum {
    #[attr(a = #)]
    Variant1(usize, u32),

    #[attr(a = #self)]
    Variant3(usize, u32),

    #[attr(a = #self.)]
    Variant4(usize, u32),

    // valid
    #[attr(a = #self.0)]
    Variant5(usize, u32),

    // valid
    #[attr(a = #self.field1)]
    Variant6 {
        field1: u32
    },
}

fn main() {}
