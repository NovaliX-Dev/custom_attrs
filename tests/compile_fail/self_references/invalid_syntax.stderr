error: Expecting reference ident or `[`.
 --> tests/compile_fail/self_references/invalid_syntax.rs:6:17
  |
6 |     #[attr(a = #)]
  |                 ^

error: Expecting `.`.
 --> tests/compile_fail/self_references/invalid_syntax.rs:9:21
  |
9 |     #[attr(a = #self)]
  |                     ^

error: Expecting ident.
  --> tests/compile_fail/self_references/invalid_syntax.rs:12:22
   |
12 |     #[attr(a = #self.)]
   |                      ^

error: Value not set for `a`.
 --> tests/compile_fail/self_references/invalid_syntax.rs:7:5
  |
7 |     Variant1(usize, u32),
  |     ^^^^^^^^

error: Value not set for `a`.
  --> tests/compile_fail/self_references/invalid_syntax.rs:10:5
   |
10 |     Variant3(usize, u32),
   |     ^^^^^^^^

error: Value not set for `a`.
  --> tests/compile_fail/self_references/invalid_syntax.rs:13:5
   |
13 |     Variant4(usize, u32),
   |     ^^^^^^^^
